sonar-batch/src/main/java/org/sonar/batch/cache/DefaultProjectCacheStatus.java

```diff
@@ -19,73 +19,67 @@
  */
 package org.sonar.batch.cache;
 
-import javax.annotation.Nullable;
+import org.apache.commons.io.FileUtils;
 
-import org.sonar.batch.bootstrap.ServerClient;
 import org.sonar.home.cache.PersistentCache;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Date;
 
 public class DefaultProjectCacheStatus implements ProjectCacheStatus {
-  private static final String STATUS_PREFIX = "cache-sync-status-";
+  private static final String STATUS_FILENAME = "cache-sync-status";
   private PersistentCache cache;
-  private ServerClient client;
 
-  public DefaultProjectCacheStatus(PersistentCache cache, ServerClient client) {
+  public DefaultProjectCacheStatus(PersistentCache cache) {
     this.cache = cache;
-    this.client = client;
   }
 
   @Override
-  public void save(@Nullable String projectKey) {
+  public void save() {
     Date now = new Date();
 
     try {
-      ByteArrayOutputStream byteOutput = new ByteArrayOutputStream();
-      try (ObjectOutputStream objOutput = new ObjectOutputStream(byteOutput)) {
+      FileOutputStream fos = new FileOutputStream(getStatusFilePath().toFile());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "fos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1854)